### PR TITLE
[14.0][ADD] sale_coupon_domain_product_reward_fixed_price: add module

### PR DESCRIPTION
**Pull request by dessanhemrayev**
Bridge module between sale_coupon_reward_fixed_price and sale_coupon_domain_product_discount

### CODE REVIEW

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

@chienandalu what do you think? this bridge module is equivalent to https://github.com/OCA/sale-promotion/tree/14.0/sale_coupon_domain_product_discount_in_field

File: sale_coupon_domain_product_reward_fixed_price/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        price_unit = program.price_unit
+        lines = self._get_paid_order_lines()
+        if (
+            program.discount_apply_on == "specific_products"
+            and program.discount_apply_on_domain_product
+        ):
+            lines = (self.order_line - self._get_reward_lines()).filtered(
+                lambda line: program._get_valid_products(line.product_id),
+            )
+            for line in lines:
```
Comment (geomer198) [inline sale_coupon_domain_product_reward_fixed_price/models/sale_order.py]: Not need use cycle here. You must use `write` method.
`lines.write({"price_unit": price_unit})`

File: sale_coupon_domain_product_reward_fixed_price/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _set_reward_fixed_price_for_lines(self, program):
+        """
+        Update discount field by program
+        """
+        price_unit = program.price_unit
+        lines = self._get_paid_order_lines()
```
Comment (geomer198) [inline sale_coupon_domain_product_reward_fixed_price/models/sale_order.py]: In this line use `_get_paid_order_lines` and than never used. For what it's here?

**Review by geomer198 — COMMENTED** [review]
Please fix comments.

File: sale_coupon_domain_product_reward_fixed_price/models/sale_order.py (hunk 1 above)
Comment (dessanhemrayev) [reply sale_coupon_domain_product_reward_fixed_price/models/sale_order.py]: Thanks, I'll fix it)

File: sale_coupon_domain_product_reward_fixed_price/models/sale_order.py (hunk 2 above)
Comment (dessanhemrayev) [reply sale_coupon_domain_product_reward_fixed_price/models/sale_order.py]: Thanks, I'll fix it)

File: sale_coupon_domain_product_reward_fixed_price/models/sale_order.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _set_reward_fixed_price_for_lines(self, program):
+        """
+        Update discount field by program
+        """
+        price_unit = program.price_unit
```
Comment (geomer198) [inline sale_coupon_domain_product_reward_fixed_price/models/sale_order.py]: For what you create separate variable and use use this variable only in condition and only once.

**Review by geomer198 — COMMENTED** [review]
Please fix comment.

**Review by geomer198 — APPROVED** [review]

**Comment by francesco-ooops** [comment]
@pedrobaeza can we merge? :)

**Comment by pedrobaeza** [comment]
Merging blindly:

/ocabot merge nobump

**Pull request merged**